solr/solrj-zookeeper/src/java/org/apache/solr/common/cloud/CollectionPropertiesZkStateReader.java

```diff
@@ -154,8 +150,14 @@ public Map<String, String> getCollectionProperties(final String collection, long
   @Override
   public void close() {
     this.closed = true;
-    notifications.shutdownNow();
-    ExecutorUtil.shutdownAndAwaitTermination(notifications);
+    if (cacheCleanerThread != null) {
+      cacheCleanerThread.interrupt();
+      try {
+        cacheCleanerThread.join();
+      } catch (InterruptedException e) {
+        Thread.currentThread().interrupt();
```

Comment:
I would say don't set interruption during closing.  Add a comment with the rationale like "ignore since we are closing"